src/app/share-preview/components/share-preview/share-preview.component.html

```diff
@@ -120,7 +126,13 @@
       <strong *ngIf="isLinkShare && isLoggedIn && hasRequested">Your request is awaiting approval from {{shareAccount.fullName}}.</strong>
       <strong *ngIf="isRelationshipShare && isLoggedIn">Your current archive does not have access to this content. Switch to view or edit.</strong>
       <div *ngIf="isLinkShare && isLoggedIn">
-        <button class="btn btn-alternate" (click)="onRequestAccessClick()" [disabled]="waiting || hasRequested" *ngIf="isLinkShare">{{ hasRequested ? 'Access requested' : requestAccessText }}</button>
+        <button class="btn btn-alternate" (click)="onRequestAccessClick()" [disabled]="waiting || hasRequested" *ngIf="isLinkShare">
+          <ng-container *ngIf="hasRequested ; else accessText">Access Requested</ng-container>
+          <ng-template #accessText>
+            <ng-container *ngIf="isAutoApprove ; else requestAccess">Accept Share</ng-container>
+            <ng-template #requestAccess>Request Access</ng-template>
```

Comment:
Oh, and I actually hadn't noticed before that we have this same code twice! Two times is right on the border, so I'll ask and defer to your judgement: would it make sense to extract a helper component or something to avoid this duplication, or would that be much more complexity than is saved?

Reply:
If this were React I'd definitely extract it out into a helper component because that would take like 5 seconds.

A complaint I have about Angular, or at least the way our Angular codebase is structured, is that creating a new component feels like an incredibly heavy operation (which is why they have the Angular CLI to help out with the boilerplate). So I personally didn't want to do run `ng generate` just for this code here.

Reply:
Oh yep, that sounds like much more work than is merited here. Unfortunate, but thanks for the explanation!